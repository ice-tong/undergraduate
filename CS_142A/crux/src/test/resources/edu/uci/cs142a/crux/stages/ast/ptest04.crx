// test use after scope

func main() : void {
    if (true) {
        var x: int;
        ::printInt(x);
    } else {
        var y: int;
        ::printInt(y);
    }
    ::printInt(x);
    ::printInt(y);

    while (true) {
        var z: int;
        ::printInt(z);
    }
    ::printInt(z);
}
