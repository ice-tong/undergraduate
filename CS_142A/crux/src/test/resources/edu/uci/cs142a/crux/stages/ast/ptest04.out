ResolveSymbolError(11,15)[Could not find x.]
ResolveSymbolError(12,15)[Could not find y.]
ResolveSymbolError(18,15)[Could not find z.]
